libs/vkd3d/utils.c

```diff
@@ -22,15 +22,43 @@
 
 #include <errno.h>
 
+#define MAX_DXGI_FORMAT DXGI_FORMAT_SAMPLER_FEEDBACK_MIP_REGION_USED_OPAQUE
+
 #define COLOR         (VK_IMAGE_ASPECT_COLOR_BIT)
 #define DEPTH         (VK_IMAGE_ASPECT_DEPTH_BIT)
 #define STENCIL       (VK_IMAGE_ASPECT_STENCIL_BIT)
 #define DEPTH_STENCIL (VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT)
 #define TYPELESS      VKD3D_FORMAT_TYPE_TYPELESS
 #define SINT          VKD3D_FORMAT_TYPE_SINT
 #define UINT          VKD3D_FORMAT_TYPE_UINT
-static const struct vkd3d_format vkd3d_formats[] =
+
+#define EMPTY {0},
+#define EMPTY0
+#define EMPTY1 EMPTY
+#define EMPTY2 EMPTY1 EMPTY
+#define EMPTY3 EMPTY2 EMPTY
+#define EMPTY4 EMPTY3 EMPTY
+#define EMPTY5 EMPTY4 EMPTY
+#define EMPTY6 EMPTY5 EMPTY
+#define EMPTY7 EMPTY6 EMPTY
+#define EMPTY8 EMPTY7 EMPTY
+#define EMPTY9 EMPTY8 EMPTY
+#define EMPTY00
+#define EMPTY10 EMPTY9 EMPTY
+#define EMPTY20 EMPTY10 EMPTY10
+#define EMPTY30 EMPTY20 EMPTY10
+#define EMPTY40 EMPTY30 EMPTY10
+#define EMPTY50 EMPTY40 EMPTY10
+#define EMPTY60 EMPTY50 EMPTY10
+#define EMPTY70 EMPTY60 EMPTY10
+#define EMPTY80 EMPTY70 EMPTY10
+#define EMPTY90 EMPTY80 EMPTY10
+/* creates 10 * tens + zero initialized empty array elements */
+#define UNUSED(tens, ones) EMPTY##tens##0 EMPTY##ones
```

Comment:
why do we need this mess?

Reply:
I'd rather just have a slow lookup for depth/stencil formats or store a pointer to a depth-stencil entry in the regular format lookup thingy or something than this.